cpu/utils/monitor_simulation.py

```diff
@@ -0,0 +1,51 @@
+from cosapp.drivers import NonLinearSolver
+
+from cpu.systems import CPUSystem
+
+
+def run_simulation(queue):
+    """Run simulation at each time step collecting live CPU data."""
+    cpu = CPUSystem("cpu")
+    cpu.add_driver(NonLinearSolver("solver", max_iter=10, factor=1.0, tol=1e-6))
+    cpu["exchanger.h_adder"] = 150
+    cpu["cpu.heat_capacity"] = 100
+
+    simulation_results = []
+    i = 0
+    previous_T = 0.0
+    while True:
+        data = queue.get()
+        if data == "DONE":
+            break
+
+        # prepare and run simu
+        t, usage, measured_T, fan_rpms = data
+        cpu["cpu.usage"] = usage
+        cpu["T_cpu"] = previous_T if i != 0 else measured_T
+        cpu.run_drivers()
+
+        # get results
+        simulated_T = cpu["cpu.next_T"]
+        cpu["cpu.T"] = simulated_T
+        previous_T = simulated_T
```

Comment:
can replace `previous_T` with `simulated_T` to be simpler.
```suggestion
        cpu["T_cpu"] = simulated_T if i != 0 else measured_T
        cpu.run_drivers()

        # get results
        simulated_T =  cpu["cpu.T"] = cpu["cpu.next_T"]
```

Reply:
done